json/tracing_test.go

```diff
@@ -0,0 +1,68 @@
+package json_test
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/json"
+	. "github.com/uber/tchannel-go/testutils/testtracing"
+	"golang.org/x/net/context"
+	"testing"
+)
+
+// JSONHandler tests tracing over JSON encoding
+type JSONHandler struct {
+	TraceHandler
+	t *testing.T
+}
+
+func (h *JSONHandler) firstCall(ctx context.Context, req *TracingRequest) (*TracingResponse, error) {
+	jctx := json.Wrap(ctx)
+	var response TracingResponse
+	peer := h.Ch.Peers().GetOrAdd(h.Ch.PeerInfo().HostPort)
+	err := json.CallPeer(jctx, peer, h.Ch.PeerInfo().ServiceName, "call", req, &response)
+	require.NoError(h.t, err)
+	return &response, err
+}
+
+func (h *JSONHandler) callJSON(ctx json.Context, req *TracingRequest) (*TracingResponse, error) {
+	return h.HandleCall(ctx, req,
+		func(ctx context.Context, req *TracingRequest) (*TracingResponse, error) {
+			jctx := ctx.(json.Context)
+			sc := h.Ch.Peers().GetOrAdd(h.Ch.PeerInfo().HostPort)
+			childResp := new(TracingResponse)
+			require.NoError(h.t, json.CallPeer(jctx, sc, h.Ch.PeerInfo().ServiceName, "call", req, childResp))
+			return childResp, nil
```

Comment:
the handler returns an error, but we're using `require.NoError`. Should we just return the error from the `CallPeer` instead?

Also be careful about using `require` when it's not from the main goroutine. It uses the `Fail` semantics which are only supported on the main test goroutine:
https://golang.org/pkg/testing/#T.FailNow
```
FailNow must be called from the goroutine running the test or benchmark function
```